src/menus/ViewMenus.cpp

```diff
@@ -127,33 +127,44 @@ void DoZoomFitV(AudacityProject &project)
    auto &tracks = TrackList::Get( project );
 
    // Only nonminimized audio tracks will be resized
-   auto range = tracks.Any<AudioTrack>()
+   auto resizableTracks = tracks.Any<AudioTrack>()
       - [](const Track *pTrack){
          return TrackView::Get( *pTrack ).GetMinimized(); };
-   auto count = range.size();
-   if (count == 0)
+
+   // Bug 2803: Cast to int, because otherwise the result of 
+   // division will be unsigned too, and will be a very large number
+   // if height will be negative!
+   auto resizableTracksCount = static_cast<int>(resizableTracks.size());
+
+   if (resizableTracksCount == 0)
       return;
 
    // Find total height to apportion
    auto height = viewInfo.GetHeight();
-   height -= 28;
-   
-   // The height of minimized and non-audio tracks cannot be apportioned
-   height -=
-      tracks.Any().sum( TrackView::GetTrackHeight )
-         - range.sum( TrackView::GetTrackHeight );
+   height -= static_cast<int>(tracks.size() - 1) * kSeparatorThickness 
```

Comment:
Separators are not yet shrunk to 1 pixel with this commit.  Will later work complicate this function?

Reply:
No they aren't, probably i will try to add more helper functions to handle this case